### DISCUSSION TITLE
what is a frame?

### DISCUSSION THREAD

**Discussion by pdJeeves** [stack-of-tasks/pinocchio#1883] [Q&A]
I’m looking into using pinocchio as an IK library for animations; so i’m assuming that “frame” means something in robotics; regardless, my first goal is to generate a model from an animation armature. But I can’t figure out what add a frame does, or what it is, or if i need to?  And the docs for “operational frames” just kind question if they should be explained or not; and i’m also unsure if thats the same thing as a frame on a model.

My first theory was that a joint is like an elbow and a frame is like a bone; but joints can be added to other joints with an offset, not just to frames; so that doesn’t seem to be the case, also adding a joint to a frame seems to have no settings for the joint.  

My next theory was that a frame is like a base class of a joint, because adding a frame has a previous frame argument; and if i add a joint then there is no previous frame on that joint, so the previous frame must be the joint itself?  But that doesn’t make much sense either?

Then i thought maybe it’s like a frame of reference? but then why would it have an offset? I guess a frame of reference would just be the joint matrix though so it can’t be that. 

So yeah, what is it?

**Comment by olivier-stasse** [comment]
Hi @pdJeeves pdJeeves thanks for your interest in Pinocchio.
I suggest that you walk through these tutorials on the [URDF format](http://wiki.ros.org/urdf/Tutorials).
This is a common format in robotics to describe a robot.
You should find the answers to all your questions.
